R/dbpath_build.R

```diff
@@ -0,0 +1,98 @@
+#' Build a dbpath object
+#'
+#' Builds a [dbpath()] object from its parts. All parts are optional other than
+#' `dialect`. To create a database URL, use `format()` on the resulting `dbpath`
+#' object.
+#'
+#' @examples
+#' dbpath_build("postgres", host = "localhost", port = 5432, database = "money")
+#'
+#' @param dialect,driver The SQL database `dialect` and optional R `driver`
+#'   package to use with this dialect.
+#' @param username,password Optional username and password
+#' @param host,port Optional host and port
+#' @param database The name of the database
+#' @param params A named list of query parameters or a query string. For
+#'   example, either `list(foo = "bar", name = "value")` or
+#'   `foo=bar&name=value`.
+#'
+#' @return A [dbpath()] object.
+#'
+#' @export
+dbpath_build <- function(
+  dialect,
+  driver = NULL,
```

Comment:
I didn't include defaults for any of these, but it occurred to me that you could have them look for values in envvars by default. 🤔

Reply:
I do a lot of wrapping of the SQLALchemy version of this function in tools, and the wrappers use environment variables to set the different parts! (For example, siuba's test suite has a [big config of env variables](https://github.com/machow/siuba/blob/main/siuba/tests/helpers.py#L25) to tweak how different backends get connected to).

I wonder if it's useful to keep this function dead simple and not access env variables, and let people build on top of it?

cc @chriscardillo -- do you connect via env variables for any of you [dbcooper](https://github.com/chriscardillo/dbcooper) wrapper packages?

Reply:
Yeah I think keeping it simple is a good idea. There are so many ways to store those kinds of options — envvars, R options, the config or keyring packages — that it's probably best to be flexible here.